core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -2880,48 +2883,159 @@ components:
+              $ref: "#/components/schemas/EntityAddress"
+            key_schema_hash:
               $ref: '#/components/schemas/SchemaHash'
-            local_type_index:
+            key_local_type_index:
+              $ref: '#/components/schemas/LocalTypeIndex'
+            value_schema_hash:
+              $ref: '#/components/schemas/SchemaHash'
+            value_local_type_index:
               $ref: '#/components/schemas/LocalTypeIndex'
-    InstanceTypePointer:
+    ObjectFieldStructure:
       allOf:
-        - $ref: '#/components/schemas/TypePointer'
+        - $ref: "#/components/schemas/SubstateSystemStructure"
         - type: object
           required:
-            - index
+            - value_schema
           properties:
-            index:
+            value_schema:
+              $ref: "#/components/schemas/ObjectSubstateTypeReference"
+    ObjectKeyValuePartitionEntryStructure:
+      allOf:
+        - $ref: "#/components/schemas/SubstateSystemStructure"
+        - $ref: "#/components/schemas/KeyValueBasedStructure"
+    ObjectIndexPartitionEntryStructure:
+      allOf:
+        - $ref: "#/components/schemas/SubstateSystemStructure"
+        - $ref: "#/components/schemas/KeyValueBasedStructure"
+    ObjectSortedIndexPartitionEntryStructure:
+      allOf:
+        - $ref: "#/components/schemas/SubstateSystemStructure"
+        - $ref: "#/components/schemas/KeyValueBasedStructure"
+    KeyValueBasedStructure:
+      type: object
+      required:
+        - key_schema
+        - value_schema
+      properties:
+        key_schema:
+          $ref: "#/components/schemas/ObjectSubstateTypeReference"
+        value_schema:
+          $ref: "#/components/schemas/ObjectSubstateTypeReference"
+    SystemFieldKind:
+      type: string
+      enum:
+        - TypeInfo
+    ObjectSubstateTypeReferenceType:
+      type: string
+      enum:
+        - Package
+        - ObjectInstance
+    ObjectSubstateTypeReference:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          $ref: "#/components/schemas/ObjectSubstateTypeReferenceType"
+      discriminator:
+        propertyName: type
+        mapping:
+          # NOTE: These need to match ObjectSubstateTypeReferenceType
+          Package: '#/components/schemas/PackageObjectSubstateTypeReference'
+          ObjectInstance: '#/components/schemas/ObjectInstanceTypeReference'
+    PackageObjectSubstateTypeReference:
+      allOf:
+        - $ref: "#/components/schemas/ObjectSubstateTypeReference"
+        - $ref: "#/components/schemas/PackageTypeReference"
+    ObjectInstanceTypeReference:
+      allOf:
+        - $ref: "#/components/schemas/ObjectSubstateTypeReference"
+        - type: object
+          required:
+            - entity_address
+            - schema_hash
+            - instance_type_index
+            - local_type_index
+          properties:
+            entity_address:
+              $ref: "#/components/schemas/EntityAddress"
```

Comment:
_(note from author)_

I noticed that we sometimes map address/NodeId into `EntityAddress` (a bech32m string) and sometimes to `EntityReference` (essentially a `(entity_type, is_global, EntityAddress)` tuple).
I wonder: what is the rule?